internal/service/budgets/budget_data_source.go

```diff
@@ -0,0 +1,345 @@
+							Computed: true,
+						},
+						"include_recurring": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_refund": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_subscription": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_support": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_tax": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"include_upfront": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"use_amortized": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"use_blended": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name_prefix": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"notification": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"comparison_operator": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"notification_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"subscriber_email_addresses": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"subscriber_sns_topic_arns": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"threshold": {
+							Type:     schema.TypeFloat,
+							Computed: true,
+						},
+						"threshold_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"planned_limit": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"amount": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"start_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"unit": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"time_period_end": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"time_period_start": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"time_unit": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"budget_exceeded": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+const (
+	DSNameBudget = "Budget Data Source"
+)
+
+func dataSourceBudgetRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).BudgetsConn()
+
+	accountID, budgetName, err := BudgetParseResourceID(d.Id())
```

Comment:
The data source won't have an ID set yet like a resource would on Read. By making `name` and required and `account_id` optional/computed, those values should be enough to call the finder.

Reply:
Ah ok. So I can just remove 270-273 and just call the finder on line 275?

Reply:
Yes - though you may need to add a step that merges `name` and `name_prefix` prior to calling the finder. 

Ref: https://github.com/hashicorp/terraform-provider-aws/pull/31691/files#r1214659786